parser/parser.y

```diff
@@ -2943,6 +2943,14 @@ FunctionCallKeyword:
 	{
 		$$ = &ast.FuncCallExpr{FnName: model.NewCIStr($1), Args: $3.([]ast.ExprNode)}
 	}
+|	"DATE" stringLit
+	{
+		tp := types.NewFieldType(mysql.TypeString)
```

Comment:
The return fieldType should be mysql.TypeDate,
and charset, collate be binary,
flen, and decimal be 10 and 0 respectively.
mysql> select date'2017/12/12';
Field   1:  `date'2017/12/12'`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       DATE
Collation:  binary (63)
Length:     10
Max_length: 10
Decimals:   0
Flags:      NOT_NULL BINARY


+------------------+
| date'2017/12/12' |
+------------------+
| 2017-12-12       |
+------------------+
1 row in set (0.00 sec)